### PR TITLE
ADD section about Orion performance tuning

### PR DESCRIPTION
**Pull request by fgalan**
Appart of providing content to the performance tunning documentation, this PR include some minor additions:

* Changing default cache sync period to 60 seconds (mentioned in CNR)
* Explaining how throtling will work in multi-CB
* A new test for -statNotifQueue

*Hint*: it could be better to review the perf_tunning.md document in rendered view: https://github.com/telefonicaid/fiware-orion/blob/hardening/doc_tunning/doc/manuals/admin/perf_tuning.md

### CODE REVIEW

File: doc/manuals/admin/logs.md [hunk 1]
```diff
@@ -16,6 +16,14 @@ When starting the Orion context broker, if a previous log file exist:
 -   If **-logAppend** is not used, then the existing file is renamed,
     appending the text ".old" to its name.
 
+The `-logLevel` option allows to chose which error messages will be printed in the log:
```
Comment (fgalan) [inline doc/manuals/admin/logs.md]: Strictly speaking, `-logLevel` is not part of this PR, but I have added this in order to advance a little bit with documentation.

File: doc/manuals/admin/logs.md [hunk 2]
```diff
@@ -61,15 +69,15 @@ The different fields in each line are as follows:
...
     line was generated.
 -   **lvl (level)**. There are four level types:
-    -   INFO: This level designates informational messages that
-        highlight the progress of Orion.
-    -   WARNING: This level designates potentially harmful situations.
-        There is a minor problem that should be fixed.
     -   ERROR: This level designates error events. There is a severe
-        problem that should be fixed.
-    -   FATAL: This level designates very severe error events that will
+        problem that should be fixed. A subclass of ERROR is FATAL,
```
Comment (fgalan) [inline doc/manuals/admin/logs.md]: Old typo, that have survived quite a bit... :)

File: doc/manuals/user/walkthrough_apiv1.md [hunk 3]
```diff
@@ -1025,8 +1025,12 @@ Let's examine in detail the different elements included in the payload:
     as in the example above, makes a notification not to be sent
     if a previous notification was sent less than 5 seconds ago, no
     matter how many actual changes take place in that period. This is give the 
-   notification receptor a means to protect itself against context producers 
-   that update attribute values too frequently.
+    notification receptor a means to protect itself against context producers
```
Comment (fgalan) [inline doc/manuals/user/walkthrough_apiv1.md]: Suggestions on how to explain this cleaner are welcome (please provide suggested wording).

File: doc/manuals/admin/logs.md (hunk 1 above)
Comment (fgalan) [inline doc/manuals/admin/logs.md]: NTC

File: doc/manuals/admin/logs.md (hunk 2 above)
Comment (fgalan) [inline doc/manuals/admin/logs.md]: NTC

File: doc/manuals/admin/cli.md [hunk 4]
```diff
@@ -113,17 +113,8 @@ The list of available options is the following:
...
-    internal logic module at the same time), "read" (which ensures that
-    in a given CB node as much as 1 read request is being processed by
-    the internal logic module at the same time, write request can
-    execute concurrently), "write" (which ensures that in a given CB
-    node as much as 1 write request is being processed by the internal
-    logic module at the same time, write request can
-    execute concurrently) and "none" (which allows all the requests
-    being executed concurrently). Default value is "all". For
-    Active-Active Orion configuration "none" is recommended.
+    mutext policy. See [performance tuning](perf_tuning.md#mutex-policy-impact-in-performance) documentation
```
Comment (kzangeli) [inline doc/manuals/admin/cli.md]: mutext -> mutex :-)

File: doc/manuals/admin/extra/indexes_analysis.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
+# Database indexes analysis
+
+The following analyzis shows the TPS (transation per second) and storage
```
Comment (kzangeli) [inline doc/manuals/admin/extra/indexes_analysis.md]: analyzis -> analysis

File: doc/manuals/admin/extra/indexes_analysis.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+consumption figures for different indexes configuration and number of
```
Comment (kzangeli) [inline doc/manuals/admin/extra/indexes_analysis.md]: ... figures for configurations with different indexes and different numbers of entities ...

File: doc/manuals/admin/extra/indexes_analysis.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+entities in the Orion Context Broker database. We have used Orion 0.14.0
+in this analysis. Each transaction comprises one entity (either creating
+it, querying for it or updating it).
+
+Please, take into account that this information is provided only as a
+hint to guide your decision about which indexes to use in your
+particular set up, but the results in your particular environment may
+differ depending on hardware profile, the particular entities being used
+for the test, set up situation, etc. In this particular case, the
```
Comment (kzangeli) [inline doc/manuals/admin/extra/indexes_analysis.md]: set up -> setup

File: doc/manuals/admin/extra/indexes_analysis.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+resources of the system under test (a VMware-based VM) are: 2 vCPU (on a
```
Comment (kzangeli) [inline doc/manuals/admin/extra/indexes_analysis.md]: under test -> being tested

File: doc/manuals/admin/extra/indexes_analysis.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+physical host based on Intel Xeon E5620@2.40GHz) and 4GB RAM. Both Orion
+and MongoDB run in the same VM. The tool to generate load is JMeter
```
Comment (kzangeli) [inline doc/manuals/admin/extra/indexes_analysis.md]: JMeter,

File: doc/manuals/admin/extra/indexes_analysis.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+using the configuration that can be found at [the following
+location](https://github.com/telefonicaid/fiware-orion/tree/develop/test/LoadTest)
+(orionPerformanceOnlyQueries\_v2.0.jmx,
+orionPerformanceOnlyAppends\_v2.0.jmx and
+orionPerformanceAppendsAndUpdates\_v2.0.jmx) and running in a separated
```
Comment (kzangeli) [inline doc/manuals/admin/extra/indexes_analysis.md]: separated -> separate

File: doc/manuals/admin/extra/indexes_analysis.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+VM (but in the same subnet, i.e. L2 connectivity with the system under
+test.
```
Comment (kzangeli) [inline doc/manuals/admin/extra/indexes_analysis.md]: being tested

File: doc/manuals/admin/extra/indexes_analysis.md [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+
+Test cases are entity query, entity creation and mixing creation and update.
+
+Throughput:
+
+
+| Case - Indexes                                           |  10,000 entities  | 100,000 entities   | 1,000,000 entities |
+|:---------------------------------------------------------|:----------------- |:------------------ |:------------------ |
+| Query - none                                             | 115.3             | 12.2               | 2                  |
```
Comment (kzangeli) [inline doc/manuals/admin/extra/indexes_analysis.md]: 115.3 ... of what ?
I would put something in the header of the table, like:   "TPS with a db of 10,000 entities"

File: doc/manuals/admin/extra/indexes_analysis.md [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+| 100,000 entities -  `_id.type`                         | 10              | 0.052                        |
+| 100,000 entities -  separated `_id.id` and `_id.type`  | 13              | 0.067                        |
+| 100,000 entities -  compound `{_id.id`,`_id.type}`     | 12              | 0.062                        |
+| 1,000,000 entities - none (\*)                         | 124             | 0.077                        |
+| 1,000,000 entities - `_id.id`                          | 154             | 0.076                        |
+| 1,000,000 entities - `_id.type`                        | 145             | 0.073                        |
+| 1,000,000 entities - separated `_id.id` and `_id.type` | 175             | 0.088                        |
+| 1,000,000 entities - compound (`_id.id`,`_id.type}`    | 161             | 0.079                        |
+
+(\*) Althought we don't set any index, note that the MongoDB always set
```
Comment (kzangeli) [inline doc/manuals/admin/extra/indexes_analysis.md]: Although
MongoDB always sets

File: doc/manuals/admin/logs.md (hunk 1 above)
Comment (kzangeli) [inline doc/manuals/admin/logs.md]: chose -> choose

File: doc/manuals/admin/cli.md (hunk 4 above)
Comment (fgalan) [inline doc/manuals/admin/cli.md]: Fixed in 04b5167126c917d79675c9752a2477a84e36b197

File: doc/manuals/admin/extra/indexes_analysis.md (hunk 5 above)
Comment (fgalan) [inline doc/manuals/admin/extra/indexes_analysis.md]: Fixed in 311f96c

File: doc/manuals/admin/extra/indexes_analysis.md (hunk 6 above)
Comment (fgalan) [inline doc/manuals/admin/extra/indexes_analysis.md]: Fixed in 311f96c

File: doc/manuals/admin/extra/indexes_analysis.md (hunk 7 above)
Comment (fgalan) [inline doc/manuals/admin/extra/indexes_analysis.md]: Fixed in 311f96c

File: doc/manuals/admin/extra/indexes_analysis.md (hunk 8 above)
Comment (fgalan) [inline doc/manuals/admin/extra/indexes_analysis.md]: Fixed in 311f96c

File: doc/manuals/admin/extra/indexes_analysis.md (hunk 9 above)
Comment (fgalan) [inline doc/manuals/admin/extra/indexes_analysis.md]: Fixed in 311f96c

File: doc/manuals/admin/extra/indexes_analysis.md (hunk 10 above)
Comment (fgalan) [inline doc/manuals/admin/extra/indexes_analysis.md]: Fixed in 311f96c

File: doc/manuals/admin/extra/indexes_analysis.md (hunk 11 above)
Comment (fgalan) [inline doc/manuals/admin/extra/indexes_analysis.md]: Fixed in 311f96c

File: doc/manuals/admin/extra/indexes_analysis.md (hunk 12 above)
Comment (fgalan) [inline doc/manuals/admin/extra/indexes_analysis.md]: Fixed in 311f96c (TPS added to the title of the table, doing it int the column would be more costly)

File: doc/manuals/admin/extra/indexes_analysis.md (hunk 13 above)
Comment (fgalan) [inline doc/manuals/admin/extra/indexes_analysis.md]: Fixed in 311f96c

**Comment by kzangeli** [comment]
LGTM
